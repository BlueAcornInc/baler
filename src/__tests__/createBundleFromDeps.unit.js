const getModuleUnderTest = () => {
    jest.resetModules();
    return require('../createBundleFromDeps');
};
const stubDateISOString = timestamp => {
    // We include a timestamp in the generated bundles. Force it to be
    // consistent so snapshot tests work as expected
    jest.spyOn(global.Date.prototype, 'toISOString').mockImplementation(
        jest.fn().mockReturnValue(timestamp),
    );
};
const stubFilePaths = mappings => {
    jest.mock('../fsPromises', () => ({
        readFile: async path => {
            if (mappings[path]) {
                return mappings[path];
            }

            throw new Error(`Failed reading ${path} in mock fs`);
        },
    }));
};

test('bundles a simple require app with no config', async () => {
    const files = {
        'foo.js': `
            define(['bar'], function(bar) {
                console.log(bar);
            });
        `,
        'bar.js': `
            define([], function() {
                return 'bar';
            });
        `,
    };
    stubFilePaths(files);
    stubDateISOString(
        'Mon Sep 16 2019 17:46:16 GMT-0500 (Central Daylight Time)',
    );
    const { createBundleFromDeps } = getModuleUnderTest();

    const results = await createBundleFromDeps(
        'core-bundle',
        ['foo', 'bar'],
        '',
        {},
        'Vendor/theme',
    );

    expect(results.bundle).toMatchSnapshot();
    expect(results.bundleFilename).toBe('core-bundle.js');
    expect(typeof results.map).toBe('string');
});
