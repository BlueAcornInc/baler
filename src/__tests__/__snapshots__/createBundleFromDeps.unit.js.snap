// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundles a simple require app with no config 1`] = `
"/* Generated by @magento/baler - Mon Sep 16 2019 17:46:16 GMT-0500 (Central Daylight Time) */
            define('foo', ['bar'], function(bar) {
                console.log(bar);
            });
        

            define('bar', [], function() {
                return 'bar';
            });
        "
`;
